share/spack/setup-env.fish

```diff
@@ -372,7 +372,14 @@ end
 
 
 function spack_runner -d "Runner function for the `spack` wrapper"
-
+    # Store LD_LIBRARY_PATH variables from spack shell function
+    # This is necessary because MacOS System Integrity Protection clears
+    # variables that affect dyld on process start.
+    for var in LD_LIBRARY_PATH DYLD_LIBRARY_PATH DYLD_FALLBACK_LIBRARY_PATH
+        if set -q $var
+            set -gx SPACK_$var $$var
```

Comment:
Crazy how much easier this is in fish than sh

Reply:
Can you remove `LD_LIBRARY_PATH` while you're at it? Only DYLD_* vars need to be "saved".

Reply:
Hm, let me submit a separate PR for it.